integration/cluster_test.go

```diff
@@ -371,3 +375,48 @@ func clusterMustProgress(t *testing.T, membs []*member) {
 		mcancel()
 	}
 }
+
+func TestStopTriggerTransferLeader(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := NewClusterV3(t, &ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	leaderIdx1 := clus.waitLeader(t, clus.Members)
+	leaderID1 := clus.Members[leaderIdx1].s.ID()
+
+	donec1 := make(chan struct{})
+	donec2 := make(chan struct{})
+	go func() {
+		defer close(donec2)
+
+		// fail when the leader is taken down without transfer
+		// and succeed if transfer works
+		for {
+			select {
+			case <-donec1:
+				return
+			default:
+				kvc := toGRPC(clus.Client((leaderIdx1 + 1) % 3)).KV
+				_, err := kvc.Range(context.TODO(), &etcdserverpb.RangeRequest{Key: []byte("foo")})
+				if err != nil && grpc.ErrorDesc(err) != transport.ErrConnClosing.Desc { // when follower lost leader
```

Comment:
Now it returns `cluster_test.go:403: rpc error: code = 13 desc = etcdserver: request timed out, possibly due to previous leader failure`.

Would it be better just to `time.Sleep` before we start `range` requests?